pkg/ibmvpc/tgw_analysis.go

```diff
@@ -12,23 +12,56 @@ const (
 	denyAction   = "deny"
 )
 
-// given a TransitConnection object, analyze its prefix filters, and get the permitted set of subnets
-// from the VPC to be connected via the TGW
-func getTransitConnectionFiltersForVPC(tc *datamodel.TransitConnection, vpc *VPC) (map[string]bool, error) {
-	res := map[string]bool{}
+// getVPCdestSubnetsByAdvertisedRoutes returns a slice of subnets from vpc, which can be destinations of the
+// transit gateway tg, based on its available routes (as determined by prefix filters and matched address prefixes)
+func getVPCdestSubnetsByAdvertisedRoutes(tg *TransitGateway, vpc *VPC) (res []*Subnet) {
 	for _, subnet := range vpc.subnets() {
-		matched, err := isSubnetMatchedByPrefixFilters(subnet, tc)
+		if isSubnetTGWDestination(tg, subnet) {
+			res = append(res, subnet)
+		}
+	}
+	return res
+}
+
+func isSubnetTGWDestination(tg *TransitGateway, subnet *Subnet) bool {
+	dstIPB := common.NewIPBlockFromCidrOrAddress(subnet.cidr)
+	for _, routesListPerVPC := range tg.availableRoutes {
```

Comment:
we could, though in case of potential overlapping address prefixes, there could be routing to another VPC (may depend on prefix filters and/or conflict resolution aspects). For now will restrict to only available routes for the VPC this subnet comes from.